crates/ruma-events-macros/src/event_parse.rs

```diff
@@ -194,45 +205,43 @@ pub struct EventEnumInput {
 
     /// An array of valid matrix event types.
     ///
-    /// This will generate the variants of the event type "name". There needs to be a corresponding
+    /// This will generate the variants of the event type "kind". There needs to be a corresponding
     /// variant in `ruma_events::EventType` for this event (converted to a valid Rust-style type
     /// name by stripping `m.`, replacing the remaining dots by underscores and then converting
     /// from snake_case to CamelCase).
     pub events: Vec<EventEnumEntry>,
 }
 
-impl Parse for EventEnumInput {
-    fn parse(input: ParseStream<'_>) -> parse::Result<Self> {
-        let attrs = input.call(Attribute::parse_outer)?;
-        // "name" field
-        let _: kw::kind = input.parse()?;
-        let _: Token![:] = input.parse()?;
-
-        // the name of our event enum
-        let name: EventKind = input.parse()?;
-        let _: Token![,] = input.parse()?;
-
-        // "events" field
-        let _: kw::events = input.parse()?;
-        let _: Token![:] = input.parse()?;
-
-        // an array of event names `["m.room.whatever", ...]`
-        let content;
-        bracketed!(content in input);
-        let events = content.parse_terminated::<_, Token![,]>(EventEnumEntry::parse)?;
-        let events = events.into_iter().collect();
-
-        Ok(Self { attrs, name, events })
-    }
-}
-
-pub struct EventEnumEntry {
-    pub attrs: Vec<Attribute>,
-    pub ev_type: LitStr,
+/// The entire `event_enum!` macro structure directly as it appears in the source code.
+pub struct EventEnumInput {
+    pub(crate) enums: Vec<EventEnumDecl>,
 }
 
-impl Parse for EventEnumEntry {
-    fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
-        Ok(Self { attrs: input.call(Attribute::parse_outer)?, ev_type: input.parse()? })
+impl Parse for EventEnumInput {
+    fn parse(input: ParseStream<'_>) -> parse::Result<Self> {
+        let mut enums = vec![];
+        while !input.is_empty() {
+            let attrs = input.call(Attribute::parse_outer)?;
+
+            if input.parse::<Token![enum]>().is_err() {
+                break;
+            };
```

Comment:
Instead of ending parsing, unexpected input should cause an error.

```suggestion
            let _: Token![enum] = input.parse()?;
```